hw/top_earlgrey/data/top_earlgrey.hjson

```diff
@@ -8,11 +8,51 @@
 
   datawidth: "32",  # 32-bit datawidth
 
-  clocks: [
-    { name: "main", freq: "100000000" }
-    { name: "fixed", freq: "100000000" }
-    { name: "usb", freq: "48000000" }
-  ]
+  // This is the clock data strcture of the design.
+  // The src key indicates the raw clock sources in the design
+  // The groups key indicates the various clock groupings in the design
+  clocks: {
```

Comment:
@eunchan 

i'm about half-way through what you described. I think what you're saying does make sense..but i'm a bit conflicted about having information related to the module be elsewhere.  In any case, the clock info should be grouped up at least.  We can continue discussing the other piece.

Reply:
Did you already do this? super fast. Thanks for taking the concept and implement here.

Yeah we can discuss the module info later. My intention in the suggestion (bad probably), was to gather the information in one place so that users can easily recognize the structure. But as you said, it spread out the module information into multiple locations as you see in inter_module connections. But those small leaks may be inevitable.

Reply:
yeah... i know what you mean.  Would you be okay if we pushed this in first, and then iterated on your idea? We can follow-up with more PRs.  I mostly wanted to do this so I can get a model going for the clock controller and the clock split Scott was asking about.

Reply:
SG